In [1]:
# import dependencies
# data cleaning and manipulation 
import pandas as pd
import numpy as np

# machine learning
from sklearn.preprocessing import StandardScaler
import sklearn.linear_model as skl_lm
from sklearn import preprocessing
from sklearn import neighbors
from sklearn.metrics import confusion_matrix, classification_report, precision_score
from sklearn.model_selection import train_test_split

In [2]:
df = pd.read_csv('data/data.csv')

In [3]:
cols = ["id",
        "Unnamed: 32",
        'radius_worst', 
        'texture_worst', 
        'perimeter_worst', 
        'area_worst', 
        'smoothness_worst', 
        'compactness_worst', 
        'concavity_worst',
        'concave points_worst', 
        'symmetry_worst', 
        'fractal_dimension_worst',
        'perimeter_mean',
        'perimeter_se', 
        'area_mean', 
        'area_se',
        'concavity_mean',
        'concavity_se', 
        'concave points_mean', 
        'concave points_se']
df = df.drop(cols, axis=1)
print('Shape of the dataframe: ', df.shape)
# verify remaining columns
df.columns

Shape of the dataframe:  (569, 13)


Index(['diagnosis', 'radius_mean', 'texture_mean', 'smoothness_mean',
       'compactness_mean', 'symmetry_mean', 'fractal_dimension_mean',
       'radius_se', 'texture_se', 'smoothness_se', 'compactness_se',
       'symmetry_se', 'fractal_dimension_se'],
      dtype='object')

# Random Forest model

The Random Forest model is not a distance based technique, hence feature scaling is not required.

In [4]:
y = df.diagnosis.values
x_data = df.drop(['diagnosis'], axis = 1)

In [5]:
#Split the train and test dataset
x_train, x_test, y_train, y_test = train_test_split(x_data,y,test_size = 0.3 ,random_state=40)

In [10]:
# Random Forest Classification
from sklearn.ensemble import RandomForestClassifier
rf = RandomForestClassifier(n_estimators = 1000, random_state = 1)
rf.fit(x_train, y_train)

acc = rf.score(x_test, y_test)*100
print("Random Forest Algorithm Accuracy Score : {:.2f}%".format(acc))

Random Forest Algorithm Accuracy Score : 93.57%


### The Random Forest Model has an accuracy of 93.57%